source/AI.cpp

```diff
@@ -3702,11 +3702,23 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 			&& !(autoPilot | activeCommands).Has(Command::LAND | Command::JUMP | Command::FLEET_JUMP | Command::BOARD)
 			&& (!target || target->GetGovernment()->IsEnemy()))
 		AutoFire(ship, firingCommands, false);
+
+	if(Preferences::Has("alt-mouse turning") && !ship.IsBoarding() && !ship.IsReversing())
+	{
+		Angle shipAngle = ship.Facing();
+
+		double angDiff = player.MouseAngle().Degrees() - shipAngle.Degrees();
+		if(abs(angDiff) > 1)
+			command.SetTurn((angDiff / abs(angDiff)) * pow(-1, (floor(abs(angDiff) / 180))));
+		else
+			command.SetTurn(0);
+	}
+
```

Comment:
Since it looks like the mouse angle isn't actually important until this point, how about just storing the mouse position as a Point (relative to the centre of the screen) and doing a dot product here to figure out how well aligned the ship is to the mouse vector?



```suggestion

	if(Preferences::Has("alt-mouse turning") && !ship.IsBoarding() && !ship.IsReversing())
	{
		Point shipAngle = ship.Unit();

		double angDiff = player.MouseAngle().Unit().Dot(shipAngle);
		double mag = abs(angDiff);
		if(mag > 1.)
			command.SetTurn(angDiff > 0. ? 1. : -1.);
		else
			command.SetTurn(0);
	}

```

`player.MouseAngle().Unit()` should probably become `player.MouseUnit()`, and just store the unit vector in the direction of hte cursor.